editor/editor_inspector.cpp

```diff
@@ -3720,6 +3720,8 @@ void EditorInspector::_check_meta_name(String name) {
 		error = TTR("Invalid metadata identifier.");
 	} else if (object->has_meta(name)) {
 		error = TTR("Metadata already exists.");
+	} else if (name[0] == '_') {
+		error = TTR("Editor-only metadata only allowed setting in code.");
```

Comment:
Use a more informative error message:

```suggestion
		error = TTR("Names starting with _ are reserved for editor-only metadata.");
```